shared/docs/PROTOCOL_STATE_HASH.md

```diff
@@ -0,0 +1,181 @@
+# State Hash <!-- omit in toc -->
+
+This document describes the cross-module communication using the interfaces in [../shared/modules](../shared/modules) to compute a new state hash. See module specific documentation & implementation details inside each module respectively.
+
+- [Context Management](#context-management)
+- [Block Application](#block-application)
+
+_NOTE: The diagrams below use some [Hotstuff specific](https://arxiv.org/abs/1803.05069) terminology as described in the [HotPOKT Consensus Specification](https://github.com/pokt-network/pocket-network-protocol/tree/main/consensus) but can be adapted to other BFT protocols as well._
+
+<!-- See if there's an answer in this question to add links to notes: https://stackoverflow.com/questions/74103729/adding-hyperlinks-to-notes-in-mermaid-sequence-diagrams -->
+
+## Context Management
+
+The `Utility` and `Persistence` modules maintain a context (i.e. an ephemeral states) driven by the `Consensus` module that can be `released & reverted` (i.e. the block is invalid / no Validator Consensus reached) or can be `committed & persisted` to disk (i.e. the block is finalized).
+
+On every round of every height:
+
+1. The `Consensus` module handles a `NEWROUND` message
+2. A new `UtilityContext` is initialized at the current height
+3. A new `PersistenceRWContext` is initialized at the current height
+4. The [Block Application](#block-application) flow commences
+
+```mermaid
+sequenceDiagram
+    title Steps 1-4
+    participant B as Bus
+    participant C as Consensus
+    participant U as Utility
+    participant P as Persistence
+
+    %% Handle New Message
+    B-->>C: HandleMessage(NEWROUND)
+
+    %% NEWROUND
+
+    activate C
+    %% Create Contexts
+    C->>+U: NewContext(height)
+    U->>+P: NewRWContext(height)
+    P->>-U: PersistenceContext
+    U->>U: store context<br>locally
+    activate U
+    deactivate U
+    U->>-C: UtilityContext
+    C->>C: store context<br>locally
+    deactivate C
+
+    %% Apply Block
+    Note over C, P: 'Block Application'
+```
+
+---
+
+_The **Proposer** drives the **Validators** to agreement via the **Consensus Lifecycle** (i.e. HotPOKT)_
+
+---
+
+5. The `Consensus` module handles the `DECIDE` message
+6. The final `quorumCertificate` is propagated to the `UtilityContext` & `PersistenceContext` on `Commit`
```

Comment:
In the issue description and also here, you talk about the "final quorumCertificate" but this exact same language is not used in the [Specification](https://github.com/pokt-network/pocket-network-protocol/tree/main/consensus). Emphasis on the word **final**. By intuition, I assume we are talking about the `CommitQC`. Am I correct?

If I am correct, do you reckon it would be worth it changing the wording accordingly here?

Reply:
Great point! Updated